pkg/pillar/containerd/oci.go

```diff
@@ -332,6 +335,7 @@ func (s *ociSpec) UpdateFromVolume(volume string) error {
 			return err
```

Comment:
As a matter of fact, I'm sort of confuse here as well, since I don't think `updateFromImageConfig` has any chance of manipulating Mounts

So the question then becomes -- what exactly are you protecting it from by making a copy?

Reply:
It is question of sequence in case of previously saved json from [here](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/containerd/oci.go#L123). Which one comes after updateMounts. So we have:

1. [setupSpec](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/hypervisor/kvm.go#L416) with [UpdateFromVolume](https://github.com/lf-edge/eve/blob/0f25acd7b4772aca3159ccef4dd169312156fee4/pkg/pillar/hypervisor/containerd.go#L70) (doing nothing on the first call, no saved json) and [UpdateMounts](https://github.com/lf-edge/eve/blob/0f25acd7b4772aca3159ccef4dd169312156fee4/pkg/pillar/hypervisor/containerd.go#L74) inside
2. We have [addLoader](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/hypervisor/kvm.go#L420) which one [save](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/containerd/oci.go#L123) json to file.
3. All works fine. But when we have purge and re-run, we go to the step 1, where we have saved config which one loaded on the step of [UpdateFromVolume](https://github.com/lf-edge/eve/blob/0f25acd7b4772aca3159ccef4dd169312156fee4/pkg/pillar/hypervisor/containerd.go#L70) inside [this step](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/containerd/oci.go#L325) with old mounts inside.

Reply:
Well, if we will not call save anywhere, it will solve the problem. Let me check this variant, thank you for the point out of it